src/components/dialogs/csv-import-filter-creation-dialog.js

```diff
@@ -0,0 +1,244 @@
+/**
+ * Copyright (c) 2022, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+import Dialog from '@mui/material/Dialog';
+import DialogTitle from '@mui/material/DialogTitle';
+import DialogContent from '@mui/material/DialogContent';
+import DialogActions from '@mui/material/DialogActions';
+import { useCSVReader } from 'react-papaparse';
+import Button from '@mui/material/Button';
+import React, { useState } from 'react';
+import Grid from '@mui/material/Grid';
+import { FormattedMessage, useIntl } from 'react-intl';
+import CsvDownloader from 'react-csv-downloader';
+import { createFilter } from '../../utils/rest-api';
+import { FilterType } from '../../utils/elementType';
+import { useSelector } from 'react-redux';
+import Alert from '@mui/material/Alert';
+import { equipmentsDefinition } from './generic-filter-dialog';
+import PropTypes from 'prop-types';
+
+const CsvImportFilterCreationDialog = ({ name, onClose, open, title }) => {
+    const [createFilterErr, setCreateFilterErr] = React.useState('');
+    const intl = useIntl();
+    const { CSVReader } = useCSVReader();
+    const [value, setValue] = useState([]);
+    const activeDirectory = useSelector((state) => state.activeDirectory);
+
+    const data = [
+        [
+            intl.formatMessage({ id: 'equipmentID' }),
+            intl.formatMessage({ id: 'equipmentType' }),
+            intl.formatMessage({ id: 'distributionKey' }),
+        ],
+    ];
+
+    const csvData = () => {
+        let newData = [...data];
+        for (let i = 0; i < 10; i++) {
+            newData = [...newData, ...[[]]];
+        }
+        newData = [
+            ...newData,
+            ...[
+                [
+                    intl.formatMessage({ id: 'CSVFileComment' }),
+                    equipmentsDefinition.LINE.type,
+                ],
+            ],
+        ];
+        Object.entries(equipmentsDefinition)
+            .filter((val) => val[0] !== equipmentsDefinition.LINE.type)
+            .forEach((value) => {
+                newData = [...newData, ...[['', value[0]]]];
+            });
+        return newData;
+    };
+
+    const resetDialog = () => {
+        setValue([]);
+        setCreateFilterErr('');
+    };
+
+    const handleClose = () => {
+        onClose();
+        resetDialog();
+    };
+
+    const validateCsvFile = (rows, equipmentType) => {
+        if (rows.length === 0) {
```

Comment:
Maybe we should check that the header is the expected on at least.

Reply:
ok. done